tests/Integration/PrestaShopBundle/TranslationIntegrationTest.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace Tests\Integration\PrestaShopBundle;
+
+use PrestaShopBundle\Translation\PrestaShopTranslatorTrait;
+use Symfony\Bundle\FrameworkBundle\Test\WebTestCase;
+
+class TranslationIntegrationTest extends WebTestCase
+{
+    use PrestaShopTranslatorTrait;
+
+    private $translator;
+
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        self::bootKernel();
+        global $kernel;
+        $kernel = self::$kernel;
+        $this->translator = self::$kernel->getContainer()->get('translator');
+    }
+
+    public function testTranslator()
+    {
+        self::assertEquals('Succesful deletion', $this->translator->trans('Succesful deletion', ['legacy' => 'htmlspecialchars'], 'Admin.Notifications.Success', null));
+        self::assertEquals('The root category of the shop myshop', $this->translator->trans('The root category of the shop %shop%', ['%shop%' => 'myshop'], 'AdminCatalogNotification', null));
+        self::assertEquals('A new order has been placed on your shop', $this->translator->trans('A new order has been placed on your shop', [], 'Admin.Navigation.Header', null));
+        self::assertEquals(
+            'This value is too short. It should have {{ 3 }} characters or more.',
+            $this->translator->trans(
+                'This value is too short. It should have {{ limit }} characters or more.',
+                ['limit' => 3],
+                '',
+                null
+            )
+        );
```

Comment:
You should add an example of a pluralized message (that's the initial point of this PR after all 😅)
- `This value is too long. It should have {{ limit }} character or less.|This value is too long. It should have {{ limit }} characters or less.` (domain `validators`)

And also an example related to this weird `sprintf` special use case:
- `Image is too large (%1$d kB). Maximum allowed: %2$d kB` (domain `AdminNavigationSearch`)
- `%d results match your query "%s".` (domain `AdminNavigationSearch`)